src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoader.cs

```diff
@@ -405,14 +518,22 @@ public Bindings(ModelLoadContext ctx, DatabaseLoader parent)
                     else
                         itemType = ColumnTypeExtensions.PrimitiveTypeFromKind(kind);
 
-                    int? sourceIndex = null;
-                    bool hasSourceIndex = ctx.Reader.ReadBoolByte();
-                    if (hasSourceIndex)
+                    int cseg = ctx.Reader.ReadInt32();
+                    Contracts.CheckDecode(cseg > 0);
```

Comment:
You are explicitly writing `0` in the `Save` now. This will fail if you loaded that model.